test/libscap/helpers/engines.cpp

```diff
@@ -63,11 +63,11 @@ void check_event_is_not_overwritten(scap_t *h) {
 	iterations = 0;
 
 	/* We save some event info to check if they are still valid after some new events */
-	uint64_t prev_ts = evt->ts;
-	uint64_t prev_tid = evt->tid;
-	uint32_t prev_len = evt->len;
-	uint16_t prev_type = evt->type;
-	uint32_t prev_nparams = evt->nparams;
+	uint64_t prev_ts = scap_event_get_ts(evt);
```

Comment:
Directly accessing `scap_evt` fields is UB because the structure is pragma packed thus fields alignment might be off.
This is somewhat similar to what @LucaGuerra did on sinsp_evt APIs.